### PR TITLE
EN-67 Conversion of files, storing files and generationg manifest

### PR DESCRIPTION
**Pull request by krminta**

### CODE REVIEW

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+			try {
+				executor.runCommand(Arrays.asList(
+						pdfConversionScript,
+						pdfFile.getAbsolutePath(),
+						outDir.getAbsolutePath()));
+			} catch (ConversionException e) {
+				logger.error("Conversion failed for file: " + pdfFile.getName() + " from record: " + record.getIdentifier());
+			}
+		} else {
+			for (File file : filesToConvert) {
```
Comment (kgronowski) [inline src/main/java/pl/psnc/dei/iiif/Converter.java]: All or nothing vs partial results approach. Should we accept that some files cannot be converted?

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	private JsonArray getSequenceJson(List<String> storedFilesIds) {
+		JsonArray canvases = new JsonArray();
+
+		for (int i = 0; i < storedFilesIds.size(); i++) {
+			String imageId = storedFilesIds.get(i);
+
+			JsonObject canvas = new JsonObject();
+			canvas.put("@id", iiifImageServerUrl + "/canvas/" + imageId);
+			canvas.put("@type", "sc:canvas");
+			canvas.put("width", "1000");
```
Comment (kgronowski) [inline src/main/java/pl/psnc/dei/iiif/Converter.java]: label*

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+		return manifest;
+	}
+
+	private JsonArray getSequenceJson(List<String> storedFilesIds) {
+		JsonArray canvases = new JsonArray();
+
+		for (int i = 0; i < storedFilesIds.size(); i++) {
+			String imageId = storedFilesIds.get(i);
+
+			JsonObject canvas = new JsonObject();
```
Comment (kgronowski) [inline src/main/java/pl/psnc/dei/iiif/Converter.java]: canvas object should be added to canvases object

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+			canvas.put("@id", iiifImageServerUrl + "/canvas/" + imageId);
+			canvas.put("@type", "sc:canvas");
+			canvas.put("width", "1000");
+			canvas.put("height", "1000");
+
+			JsonArray images = new JsonArray();
```
Comment (kgronowski) [inline src/main/java/pl/psnc/dei/iiif/Converter.java]: images object should be added to canvas object

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		sequence.put("@type", "sc:Sequence");
+		sequence.put("canvases", getSequenceJson(storedFilesIds));
+
+		return manifest;
+	}
+
+	private JsonArray getSequenceJson(List<String> storedFilesIds) {
+		JsonArray canvases = new JsonArray();
+
+		for (int i = 0; i < storedFilesIds.size(); i++) {
```
Comment (kgronowski) [inline src/main/java/pl/psnc/dei/iiif/Converter.java]: for loop can be replaced with foreach

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+				throw new ConversionException("Couldn't get file " + data.srcFileUrl.toString(), e);
+			}
+		}
+	}
+
+	private void convertFiles(ConversionDataHolder dataHolder) throws InterruptedException, IOException, ConversionImpossibleException {
+		File outDir = new File(conversionDirectory, record.getIdentifier() + "/out");
+		outDir.mkdirs();
+
+		if (dataHolder.fileObjects.get(0).srcFile.getName().endsWith("pdf")) {
```
Comment (helinek) [inline src/main/java/pl/psnc/dei/iiif/Converter.java]: Are we sure that all the calls in the chain will not give nulls?

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -32,3 +34,7 @@ server.ssl.key-store-password=password12
 server.ssl.keyStoreType=PKCS12
 server.ssl.keyAlias=fresenia
 server.ssl.enabled=true
+
+# Directory used to store temporary files from conversion process
+conversion.iiif.server.url=62.3.168.209
+conversion.directory=./die/temp
```
Comment (helinek) [inline src/main/resources/application.properties]: are you sure that you want to name the dir "die" and not "dei"?

**Pull request merged**